cranelift-codegen/src/isa/x86/abi.rs

```diff
@@ -331,71 +224,72 @@ pub fn legalize_signature(
         shared_flags,
         isa_flags,
     );
+    let mut rets2 = rets.clone();
 
-    if sig.is_multi_return() && {
-        // Even if it is multi-return, see if the return values will fit into
-        // our available return registers.
-        let (gprs_required, fprs_required) = num_return_registers_required(
-            bits,
-            sig.call_conv,
-            shared_flags,
-            isa_flags,
-            &sig.returns,
-        );
-        gprs_required > ret_regs.len() || fprs_required > ret_fpr_limit
-    } {
-        debug_assert!(!sig.uses_struct_return_param());
-
-        // We're using the first register for the return pointer parameter.
-        let mut ret_ptr_param = AbiParam {
-            value_type: args.pointer_type,
-            purpose: ArgumentPurpose::StructReturn,
-            extension: ArgumentExtension::None,
-            location: ArgumentLoc::Unassigned,
-        };
-        match args.assign(&ret_ptr_param) {
-            ArgAction::Assign(ArgumentLoc::Reg(reg)) => {
-                ret_ptr_param.location = ArgumentLoc::Reg(reg);
-                sig.to_mut().params.push(ret_ptr_param);
+    if let Some(new_returns) = legalize_args(&sig.returns, &mut rets) {
```

Comment:
(light suggestion) instead of having `rets2` (which is not very descriptive a name), could we pass `&mut rets.clone()` here, and use `rets` where we used `rets2` below?

Otherwise, could we rename `rets2` to `rets_backup` and explain in a small comment why it's needed in the first place

Reply:
(or maybe `original_rets`, so it doesn't sound too weird with respect to the other comment.)